pkg/renderer/sgml/templates.go

```diff
@@ -0,0 +1,196 @@
+	SourceParagraph         string
+	StringElement           string
+	SubscriptText           string
+	SuperscriptText         string
+	Table                   string
+	TocRoot                 string
+	TocSection              string
+	UnorderedList           string
+	VerbatimLine            string
+	VerseBlock              string
+	VerseBlockParagraph     string
+	VerseParagraph          string
+}
+
+type sgmlRenderer struct {
+	functions funcMap
+	templates *Templates
+	prepared  sync.Once
+
+	// Processed templates
+	admonitionBlock         *textTemplate
+	admonitionParagraph     *textTemplate
+	article                 *textTemplate
+	articleHeader           *textTemplate
+	blankLine               *textTemplate
+	blockImage              *textTemplate
+	boldText                *textTemplate
+	calloutList             *textTemplate
+	delimitedBlockParagraph *textTemplate
+	documentDetails         *textTemplate
+	documentAuthorDetails   *textTemplate
+	externalCrossReference  *textTemplate
+	exampleBlock            *textTemplate
+	fencedBlock             *textTemplate
+	footnote                *textTemplate
+	footnoteRef             *textTemplate
+	footnoteRefPlain        *textTemplate
+	footnotes               *textTemplate
+	inlineImage             *textTemplate
+	internalCrossReference  *textTemplate
+	invalidFootnote         *textTemplate
+	italicText              *textTemplate
+	labeledList             *textTemplate
+	labeledListHorizontal   *textTemplate
+	lineBreak               *textTemplate
+	link                    *textTemplate
+	listingBlock            *textTemplate
+	literalBlock            *textTemplate
+	manpageHeader           *textTemplate
+	manpageNameParagraph    *textTemplate
+	monospaceText           *textTemplate
+	orderedList             *textTemplate
+	paragraph               *textTemplate
+	passthroughBlock        *textTemplate
+	preamble                *textTemplate
+	qAndAList               *textTemplate
+	quoteBlock              *textTemplate
+	quoteParagraph          *textTemplate
+	sectionContent          *textTemplate
+	sectionHeader           *textTemplate
+	sectionOne              *textTemplate
+	sidebarBlock            *textTemplate
+	sourceBlock             *textTemplate
+	sourceBlockContent      *textTemplate
+	sourceParagraph         *textTemplate
+	stringElement           *textTemplate
+	subscriptText           *textTemplate
+	superscriptText         *textTemplate
+	table                   *textTemplate
+	tocRoot                 *textTemplate
+	tocSection              *textTemplate
+	unorderedList           *textTemplate
+	verbatimLine            *textTemplate
+	verseBlock              *textTemplate
+	verseBlockParagraph     *textTemplate
+	verseParagraph          *textTemplate
+}
+
+func (sr *sgmlRenderer) prepareTemplates() error {
+	t := sr.templates
+	var err error
+
+	sr.prepared.Do(func() {
+
+		sr.admonitionBlock, err = sr.newTemplate("admonition-block", t.AdmonitionBlock, err)
```

Comment:
I like very much this way of propagating the error during the template setup! 👍 
it makes the code much simpler (not need to check for the error after each call to `newTemplate` 😓 )

Reply:
It shut up go cyclo, and reduced the total number of unexecuted code paths (for coverage) as well.

Reply:
yes, I can imagine! That's great! 👍